Changes mainly focused on using the AWS SDK for Go.

pfdevilliers opened the pull request:
Biggest change is guess is way we construct the curl URL to get a specific object. I updated the README.md with an example. But the basics is:

```bash
curl http://localhost:9090/bucketname/folder1/item.txt?aws_region=eu-west-1
curl http://localhost:9090/bucketname/item.txt?aws_region=us-east-1
```

The added benefit is that we are not limited to one bucket? Not sure if this was the way you wanted to go? There is a lot of work left for the caching part, I pretty much just skipped it. The Error messaging is still problematic but I've left it as is for now.

I'll switch over to your repo after this, since I'm now a collaborator :-)

qstorey commented on README.md:
````diff
@@ -9,8 +9,11 @@ Using your AWS API credentials, S3Proxy allows you to make unauthenticated HTTP
+
 ```bash
-curl http://localhost:9090/object/key
+curl http://localhost:9090/bucket1/folder1/folder2/item.txt?aws_region=eu-west-1
````
How does the region affect the request? Is it required?

I know the bucket is created within a region, or is this more for the AWS API region that the request is made to?

pfdevilliers replied:
Buckets are suppose to be global resources and by hitting the global s3 endpoint you should be able to get to the bucket. However, I could not get this to work without initiating the client with region. I think the SigV4 signing requires a location header? To be honest I'm not sure and will have to dig further into it.

qstorey replied:
So would I be able to hit the _eu-west-1_ endpoint to retrieve an object from a bucket in _us-east-1_? If that is the case, I would rather configure the region in the options. Because being a request parameter implies that it is optional. And if the global s3 endpoint doesn't work at the moment, it will result in a poor user experience.

qstorey replied:
Also, then there can be a default region and it can be overridden by CLI parameters or config files.

qstorey commented on README.md:
```diff
@@ -39,10 +42,33 @@ If you want to make modifications to S3Proxy, clone this repo, ensure that all t
 
 # Usage
+
+It is best to use this with an AWS IAM user with a minimal access policy. See
```
Nice

qstorey commented on S3Proxy/handlers.go:
```diff
@@ -19,16 +20,13 @@ func StatusHandler(w http.ResponseWriter, req *http.Request) {
 	if err != nil {
-		http.Error(w, "404: Not Found", 404)
-		return
+		http.Error(w, err.Message, err.Code)
```
Does the function (DefaultHandler) return after http.Error?

pfdevilliers replied:
Dammit seems like I missed the return after cleaning up. Will resubmit with the return included.

qstorey commented:
Nice work on the bucket functionality. Definitely worth it, I just didn't bother when I first started with S3Proxy.

pfdevilliers commented:
Updated the diff based on PR feedback. Dropped the use of ?aws_region. The aws_region is now populated from a GetBucketLocation request. This change is dependant on a version of the AWS Go SDK greater than hash 4e1ce5e4badaf78af6bfe196c90ca4f1bceaabb7. They fixed a bug.
